apps/system/js/app_chrome.js

```diff
@@ -410,62 +417,89 @@
                         this.app.themeColor, true);
       }
     }, () => {
-      console.log('Failed to pin ' + this.app.config.url);
+      console.error('Failed to pin ' + this.app.config.url);
     });
   };
 
+  AppChrome.prototype.getSiteUrl = function() {
+    var pageUrl = this.app.config.url;
+    var webManifest = this.app.webManifest;
+    var startURL = (webManifest && this.app.webManifest.start_url) ?
+      new URL(this.app.webManifest.start_url, pageUrl) : null;
+    if (startURL) {
+      return startURL.origin + startURL.pathname;
```

Comment:
This could still fail if e.g startURL is hostname/start.html, then hostname/foo.html wouldnt be within this scope. So maybe need to get the dirname (strip back to last '/') here?